lib/types/settlements.d.ts

```diff
@@ -0,0 +1,343 @@
+        created_at: number;
+        /**
+         * List of payouts created for the instant settlement.
+         */
+        ondemand_payouts?: {
+            entity: string;
+            count: number;
+            items: Settlements.RazorpaySettlement[];
+        }
+        scheduled: boolean;
+    }  
+}
+
+declare function settlements(api: any): {
+    /**
+     * Create on-demand settlement
+     * 
+     * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#create-an-instant-settlement) for required params
+     * 
+     */
+    createOndemandSettlement(params: Settlements.RazorpayInstantSettlementCreateRequestBody): Promise<Settlements.RazorpayInstantSettlement>
+    createOndemandSettlement(params: Settlements.RazorpayInstantSettlementCreateRequestBody, callback: (err: INormalizeError | null, data: Settlements.RazorpayInstantSettlement) => void): void
+    /**
+    * Get all settlements
+    *
+    * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#fetch-all-instant-settlements) for required params
+    *
+    */
+    all(params?: RazorpayQuery): Promise<{
+        entity: string;
+        count: number;
+        items: Array<Settlements.RazorpaySettlement>;
+    }>
+    all(params: RazorpayQuery, callback: (err: INormalizeError | null, data: {
+        entity: string;
+        count: number;
+        items: Array<Settlements.RazorpaySettlement>;
+    }) => void): void
+    /**
+    * Fetches a settlement given Settlement ID
+    *
+    * @param settlementId - The unique identifier of the settlement.
+    *
+    */
+    fetch(settlementId: string): Promise<Settlements.RazorpaySettlement>
+    fetch(settlementId: string, callback: (err: INormalizeError | null, data: Settlements.RazorpaySettlement) => void): void
+    /**
+    * Fetch all demand settlements
+    *
+    * @param settlementId - The unique identifier of the settlement.
+    * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#fetch-all-instant-settlements) for required params
+    * 
+    */
+    fetchAllOndemandSettlement(params: RazorpayQuery | { 'expand[]': 'ondemand_payouts' }): Promise<{
+        entity: string;
+        count: number;
+        items: Settlements.RazorpayInstantSettlement[];
+    }>
+    fetchAllOndemandSettlement(params: RazorpayQuery | { 'expand[]': 'ondemand_payouts' }, callback: (err: INormalizeError | null, data: {
+        entity: string;
+        count: number;
+        items: Settlements.RazorpayInstantSettlement[];
+    }) => void): void
+    /**
+    * Fetch on-demand settlement by ID
+    *
+    * @param settlementId - The unique identifier of the settlement.
+    * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#fetch-instant-settlement-by-id) for required params
+    * 
+    */
+    fetchOndemandSettlementById(settlementId: string, params: { 'expand[]': 'ondemand_payouts' }): Promise<Settlements.RazorpayInstantSettlement>;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Lets mark params and 'expand[]' optional